src/jit/codegencommon.cpp

```diff
@@ -10467,11 +10488,17 @@ var_types Compiler::GetHfaType(CORINFO_CLASS_HANDLE hClass)
 unsigned Compiler::GetHfaSlots(CORINFO_CLASS_HANDLE hClass)
 {
     assert(IsHfa(hClass));
+#ifdef _TARGET_ARM_
     return info.compCompHnd->getClassSize(hClass) / TARGET_POINTER_SIZE;
+#else // _TARGET_ARM64_
+    var_types hfaType = GetHfaType(hClass);
+    return info.compCompHnd->getClassSize(hClass) / emitActualTypeSize(hfaType);
+#endif // _TARGET_ARM64_
```

Comment:
No this method needs to return the number of register need to hold the HFA argument or return value.
We could rename this method to GetHfaElemCount